test/Quake/dependency-if-bug-classical.qke

```diff
@@ -0,0 +1,136 @@
+// ========================================================================== //
+// Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                 //
+// All rights reserved.                                                       //
+//                                                                            //
+// This source code and the accompanying materials are made available under   //
+// the terms of the Apache License 2.0 which accompanies this distribution.   //
+// ========================================================================== //
+
+// RUN: cudaq-opt --add-wireset --assign-wire-indices --dep-analysis -split-input-file %s | FileCheck %s
```

Comment:
curious: why are these using `split-file`? Does the pass fall apart if there is more than 1 func.func in the module?

Reply:
It relies on `assign-wire-indices` which wants only one entrypoint.

Reply:
> It relies on `assign-wire-indices` which wants only one entrypoint.

I don't think `assign-wire-indices` prohibits multiple functions from having the `cudaq::entryPointAttrName` attribute set, does it? (My quick local test suggests that correctly assigns wires just fine.)

Reply:
Well, we don't want that pass to require that there be 1 func in the module at any rate. That'll need to be fixed if it is still an issue.

Each function need not use its own wire set operation, btw. A wire set is a global resource available to all the functions in the module.

Reply:
@bmhowe23 I don't think so, no. The requirement was added to prevent multiple kernel functions getting assigned duplicate virtual wires but then calling each other, so the virtual wires indices would no longer be unique. Function calls should be ruled out anyway, so I think it should be fine.

Reply:
@atgeller to resolve this one. (We think it's easy.)

Reply:
Resolved.